### ISSUE TITLE
xud triggers SwapTimeout error for successful swap.

### ISSUE THREAD

**Issue by raladev** [ExchangeUnion/xud#1659]
### Background

Reproduction: Rare

One of the nodes was run with arby, another one was run without arby and performed manual swaps. One of the swaps was marked as successful in the cli, but got SwapTimeOut in the logs.

[Screen](https://user-images.githubusercontent.com/29906866/84661138-93b3bb80-af22-11ea-88b3-26fb023ea713.png)

### Logs
Time of swap = 11:02:38
Swap hash = 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2

[maker_f2.zip](https://github.com/ExchangeUnion/xud/files/4780410/maker_f2.zip)
[taker_f2.zip](https://github.com/ExchangeUnion/xud/files/4780477/taker_f2.zip)

**Comment by sangaman** [comment]
Can you confirm what version of xud you were using when this happened? A similar issue was recently fixed in #1642 so versions without that commit would likely experience it.

**Comment by raladev** [comment]
> Can you confirm what version of xud you were using when this happened? A similar issue was recently fixed in #1642 so versions without that commit would likely experience it.

Commit of taker:
                "com.exchangeunion.application.revision":"614cd4d221aa8fa4393ce45779b93e3cad2c35e0",
"com.exchangeunion.image.branch": "master",
"com.exchangeunion.image.created": "2020-06-12T22:17:40Z",
"com.exchangeunion.image.revision": "1b16d21d433f23b1b74b559a83116b4521a32d5b",

Commit of maker:
"com.exchangeunion.application.revision": "95829ff6664e7b14c75e67cc0468421d95fc012f",
"com.exchangeunion.image.branch": "master",
"com.exchangeunion.image.created": "2020-06-13T02:04:50Z",
"com.exchangeunion.image.revision": "13c04a529e49ba478a5df688648287bf4f3f77eb",

Both commits were created after #1642

**Comment by kilrau** [comment]
Might be related (the other way around: cli shows failed, but logs show successful): https://github.com/ExchangeUnion/xud/issues/1654

**Comment by kilrau** [comment]
Does the swap appear in `tradehistory`?

**Comment by raladev** [comment]
> Does the swap appear in `tradehistory`?

only for taker
![Screenshot 2020-06-15 17:52:03](https://user-images.githubusercontent.com/29906866/84672372-25c2c080-af31-11ea-8b53-ef5b015dc6f5.png)

**Comment by kilrau** [comment]
in #1654 only for maker...
(difference is that there the swap actually succeeded)

I guess there is still something wrong with timers @sangaman

**Comment by sangaman** [comment]
OK, this is indeed separate from #1642 and I think is rooted in issues with connext, namely the dreaded "Connext Server Error". The behavior from xud's perspective is actually what I would expect given the events. Here's what happened best I can tell from the logs.

1. Taker requests deal from Maker. Maker is to receive ETH and send BTC.
2. Maker accepts.
3. Taker sends ETH with secret preimage.
4. Maker sees incoming ETH payment and makes outgoing BTC payment with the same hash.
5. Taker sees incoming BTC payment and settles, revealing the preimage. From the taker perspective, the swap is complete and successful, and accordingly the taker's cli marks the swap as successful.
6. Maker receives preimage from the sent BTC payment, but is not able to settle the incoming ETH payment. It's not clear to me why this is happening.
7. Maker eventually times out the swap, and it goes into swap recovery.
8. In swap recovery, the preimage is retrieved from the sent payment history and it attempts to settle the incoming ETH payment, however this fails again due to an unspecified Connext server error. I'm guessing it's likely related to a locked sqlite db again but that's just a hunch. Below are the related logs.

```
15/06/2020 11:06:42.950 [SWAPS] [34mdebug[39m: checking outgoing payment status for swap 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:06:42.977 [CONNEXT] [31merror[39m: connext server error 500: Internal Server Error
15/06/2020 11:06:42.977 [SWAPS] [31merror[39m: could not settle ETH invoice for payment 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 and preimage cdad5c957f9cd57d40ba0fd2db8527ceed9d86e231d47249f36a43c7d8199dc0, **this must be resolved manually**: 8.5 - connext server error
15/06/2020 11:06:42.978 [SWAPS] [32minfo[39m: failed swap 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
```

@erkarl Any ideas as to what might be happening?

**Comment by kilrau** [comment]
Can we be a bit more selective with logs in the future and only paste log snippets from the relevant time frames? There are multiple swaps happening...

The one starting with the maker's received swap request at `11:02:38` (referenced by @raladev in the initial post) shows connext successfully setting up the incoming htlc and informing xud about it:
```
[1592218978526] INFO  (70 on connext): Successfully pushed event CONDITIONAL_TRANSFER_CREATED_EVENT to webhook: http://xud:28887/incoming-transfer
2020-06-15T11:02:58.530Z [ConnextListener] Done processing propose install event in 774 ms
```
xud picking this up:
```
15/06/2020 11:02:58.521 [CONNEXT] [34mdebug[39m: accepting incoming transfer with rHash: 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2, units: 100000000000000000, timelock 7022 and currency ETH
15/06/2020 11:02:58.522 [SWAPS] [34mdebug[39m: Executing maker code to resolve hash
15/06/2020 11:02:58.522 [SWAPS] [34mdebug[39m: Setting SendingPayment phase for deal 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:02:58.538 [LND-BTC] [34mdebug[39m: sending payment of 244635 with hash 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 to 029a38cd99d260a767a22e4fc5c3a68aab10c4c3db6e71fffa58f70e304c016961
```
until it times out and enters swap recovery
```
15/06/2020 11:03:13.677 [SWAPS] [34mdebug[39m: deal 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 failed due to SwapTimedOut
15/06/2020 11:03:13.717 [SWAPS] [32minfo[39m: recovering swap deal 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:03:13.717 [SWAPS] [34mdebug[39m: checking outgoing payment status for swap 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:03:13.721 [SWAPS] [34mdebug[39m: swap for 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 still has pending payments and will be monitored
```
Since nothing happens in between one just assumes the taker didn't claim her htlc, but she actually did!
```
15/06/2020 11:03:01.711 [LND-BTC] [34mdebug[39m: accepted htlc for invoice 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:03:01.718 [SWAPS] [34mdebug[39m: Executing taker code to resolve hash
15/06/2020 11:03:01.721 [LND-BTC] [34mdebug[39m: settling invoice for 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
15/06/2020 11:03:01.738 [SWAPS] [34mdebug[39m: Setting PaymentReceived phase for deal 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 - preimage is cdad5c957f9cd57d40ba0fd2db8527ceed9d86e231d47249f36a43c7d8199dc0
15/06/2020 11:03:01.751 [LND-BTC] [34mdebug[39m: deleted invoice subscription for 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2
```
So no matter what happens and maker entering swap recovery is all ok, but it needs to be able to pull this preimage to settle the swap and it currently doesn't.

**Comment by kilrau** [comment]
Todo's:
- [ ] it takes 3 minutes (from `11:03:13` to `11:06:42`) for xud swap recovery to try to unlock the connext payment. Why does it take so long? Why did it not try and fail much earlier? Todo: add more logging for when lnd preimage was received, processed by xud @sangaman 

Claiming the connext eth payment fails because of a connext server error (not xud fault):
```
15/06/2020 11:06:42.977 [SWAPS] �[31merror�[39m: could not settle ETH invoice for payment 222dc469651b4eb912df2fe5a9f627a3816e66fa2f41639faf4c1326f0c8d0e2 and preimage cdad5c957f9cd57d40ba0fd2db8527ceed9d86e231d47249f36a43c7d8199dc0, **this must be resolved manually**: 8.5 - connext server error
```
- [ ] todo on xud side: once preimage of leg 1 is revealed, xud swap recovery should retry unlocking payment on leg 2 until htlc expiry block height (at the moment it retries getting the preimage but **not** retrying unlocking the payment. @sangaman 
- [ ] todo01 from connext side: check our testnet indra logs what they say about why unlocking this htlc failed @ArjunBhuptani @rhlsthrm
- [ ] todo02 from connext side: test how indra behaves if it has an unresolved htlc open to the client (payment to the client) and htlc expiry is hit. It has to go on-chain! @ArjunBhuptani @rhlsthrm
